Dockerfile

```diff
@@ -20,16 +11,49 @@ RUN apk add --update --no-cache \
     giflib-dev \
     pango-dev
 
-# Install dependencies
+WORKDIR /src
+
 COPY package*.json ./
-RUN npm install
+RUN npm ci --no-audit
+
+# -- CHECK STAGE --------------------------------
+
+FROM base AS check
+
+ARG CI
+ENV CI $CI
 
-# Copy source files
 COPY . .
+RUN npm run lint
+RUN npm test
+
+# -- BUILD STAGE --------------------------------
+
+FROM base AS build
 
-# Build project
+# Define build args & map them to environment variables
+ARG GA_TRACKING_ID
+ENV GA_TRACKING_ID $GA_TRACKING_ID
+
+# Build the project and then dispose files not necessary to run the project
+# This will make the runtime image as small as possible
+COPY . .
+RUN npx next telemetry disable > /dev/null
 RUN npm run build
+RUN npm prune --production --no-audit
+RUN rm -rf .next/cache
+
+# -- RUNTIME STAGE --------------------------------
+
+FROM node:12-alpine AS runtime
```

Comment:
Because the `base` image has all the `node_modules` with all the dev dependencies, which will make the image larger than it need to be. I'm cherry picking the `node_modules` from the last step, which was pruned to have only the prod dependencies.

Reply:
Alright, makes sense 👍